pkg/scheduler/schedule_one.go

```diff
@@ -62,7 +62,17 @@ const (
 // scheduleOne does the entire scheduling workflow for a single pod. It is serialized on the scheduling algorithm's host fitting.
 func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	podInfo := sched.NextPod()
-	defer sched.DonePod(podInfo)
+	defer func() {
+		// scheduleOne must call DonePod regardless how it returns,
+		// with one exception: when the pod is ready for PreBind, the
+		// goroutine below takes over and podInfo gets set to nil.
+		//
+		// The if check is not strictly needed (DonePod accepts nil),
+		// but checking here is a bit faster.
+		if podInfo != nil {
```

Comment:
I see how it works. However, I may prefer to pursue the semantics more _explicit_ so more readable. Like, add a variable in the beginning:

```go
breakFromSchedulingCycle := true
```

Then, before starting the bindingCycle, move this variable to true:

```go
breakFromSchedulingCycle = false
```

So here we can change the check to

```go
if breakFromSchedulingCycle {
```

Reply:
Done.